src/CommandingOfficers/CommanderCinder.java

```diff
@@ -0,0 +1,181 @@
+package CommandingOfficers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import CommandingOfficers.Modifiers.COModifier;
+import Engine.XYCoord;
+import Engine.Combat.BattleSummary;
+import Engine.GameEvents.GameEventListener;
+import Terrain.GameMap;
+import Terrain.Location;
+import Units.Unit;
+import Units.UnitModel;
+
+/*
+ * Cinder is based on getting an edge in the action economy, at the cost of unit health.
+ */
+public class CommanderCinder extends Commander
+{
+  private static final CommanderInfo coInfo = new CommanderInfo("Cinder", CommanderLibrary.CommanderEnum.CINDER);
+
+  private static final double COST_MOD_PER_BUILD = 1.2;
+
+  private HashMap<XYCoord, Integer> buildCounts = new HashMap<>();
+
+  public CommanderCinder()
+  {
+    super(coInfo);
+
+    addCommanderAbility(new SearAbility(this));
+    addCommanderAbility(new WitchFireAbility(this));
+  }
+
+  public static CommanderInfo getInfo()
+  {
+    return coInfo;
+  }
+
+  @Override
+  /** Get the list of units this commander can build from the given property type. */
+  public ArrayList<UnitModel> getShoppingList(Location buyLocation)
+  {
+    setPrices(buildCounts.get(buyLocation.getCoordinates()));
+    return super.getShoppingList(buyLocation);
+  }
+
+  /*
+   * Cinder builds units at 8HP ready to act.
+   * To compensate for the ability to continue producing units in a single turn,
+   * the cost of units increases exponentially for repeated purchases from a single property.
+   */
+  public void receiveCreateUnitEvent(Unit unit)
+  {
+    if( this == unit.CO )
+    {
+      XYCoord buildCoords = new XYCoord(unit.x, unit.y);
+      buildCounts.put(buildCoords, buildCounts.get(buildCoords) + 1);
+      setPrices(0);
+
+      unit.alterHP(-2);
+      unit.isTurnOver = false;
+    }
+  }
+
+  /*
+   * Purchase price problems reset at the start of turn.
+   */
+  @Override
+  public void initTurn(GameMap map)
+  {
+    for( Location loc : ownedProperties )
+    {
+      buildCounts.put(loc.getCoordinates(), 0);
+    }
+    setPrices(0);
+    super.initTurn(map);
+  }
+
+  public void setPrices(int repetitons)
+  {
+    for( UnitModel um : unitModels )
+    {
+      um.COcost = Math.pow(COST_MOD_PER_BUILD, repetitons);
+    }
+  }
+
+  /*
+   * Sear causes 1 mass damage to Cinder's own troops, in exchange for refreshing them.
+   */
+  private static class SearAbility extends CommanderAbility
+  {
+    private static final String SEAR_NAME = "Sear";
+    private static final int SEAR_COST = 5;
+    private static final int SEAR_WOUND = -1;
+
+    SearAbility(Commander commander)
+    {
+      super(commander, SEAR_NAME, SEAR_COST);
+    }
+
+    @Override
+    protected void perform(GameMap gameMap)
+    {
+      for( Unit unit : myCommander.units )
```

Comment:
No accommodation for units that haven't moved yet? Harsh (but in character).

Reply:
I changed that in my local copy, but forgot to push it up.
You sounded like you think the D2D is bad, so I'm inclined to push the change up.

Reply:
Then again....
It's super in character, but perhaps it shouldn't be utterly useless.
Maybe Cinder should give a buff of some kind to the units that haven't moved yet, in exchange for hurting them?

Reply:
The D2D feels stronger when you start with a decent bankroll - I think I'd have to play with it more to get a feel for how strong it is.
And sure. I dunno. I would avoid making it too complex, but I'm not sure where that line is.

Reply:
As is only proper, I picked the worst-implying thing I could.
It's also the least work, and situationally useful.